core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -307,6 +307,10 @@ class GroupCoordinator(val brokerId: Int,
         }
       }
     }
+
+    // store the group metadata without holding the group lock to avoid the potential
+    // for deadlock when the callback is invoked
+    delayedGroupStore.map(groupManager.store(_))
```

Comment:
If the types match, this can usually be written as: `delayedGroupStore.map(groupManager.store)` which avoids a bit of noise. There's one other example like this.